CHANGELOG.rst

```diff
@@ -1,3 +1,27 @@
+- Fish now asks the terminal to speak the keyboard input protocol CSI u in combination with XTerm's ``modifyOtherKeys`` and Kitty's progressive enhancements.
+  This allows to bind almost any key, including arbitrary combinations of modifiers ctrl, alt and shift.
+- There is a new syntax for specifying keys for builtin ``bind``.
+  The new syntax introduces modifier names and names for some keys that don't have an obvious and printable Unicode code point. The old syntax remains mostly supported but the new one is preferred.
+  - Existing bindings that use the new names have a different meaning now.
+    For example
+    - ``bind up 'do something'`` maps the up arrow instead of the two keys ``u`` and ``p``.
+    - ``bind ctrl-x,alt-c 'do something'`` maps a sequence of two key chords.
+  - To minimize breakage, ``KEYS`` argument to ``bind`` is parsed in the old syntax in two cases:
+    - If ``KEYS`` starts with a raw escape character (`\e`) or a raw ASCII control character (`\c`).
+    - If it contains none of ``,`` or ``-`` and it is not a named key.
+  Since ``,`` and ``-`` act as separators, they need to be given as ``comma`` and ``minus`` in ambiguous cases.
+- Specifying key names as terminfo name (``bind -k``) is deprecated and may be removed in a future version.
+- Legacy bindings that use raw escape sequences instead of named modifiers/keys continue to be included in the default configuration.
+  If your terminal supports CSI u, you can remove legacy bindings (including terminfo bindings) by setting the `fish_legacy_bindings` variable to ``false`` (on shell startup or before switching key bindings).
+- The ``fish_escape_delay_ms`` configuration knob has been obsolete (for a while I think) and has been removed.
```

Comment:
No, it was *not* obsolete.

It was in fact used to distinguish between alt+x and escape,wait,x.

Please keep it for the "legacy" bindings, until they no longer exist.